### PR TITLE
feat: improve chat endpoint response parsing

### LINKED ISSUE #8: `useTextBuffer` Issue with Chat Endpoint
I found that when using an OpenAI stream with the "chat" endpoint, we get JSON objects like this:

```js
['{"role":"assistant"}{"content":"This"}', '{"content":" is"}{"content":" an"}', '{"content":" example"}']
```

So when we use useTextBuffer or StreamingTextURL, we end up with these strings instead of plain text.

I came up with a helper function in my project that parses the content out of these objects. Happy to submit a PR for this. Maybe one can add another parameter to useTextBuffer/StreamTextURL, like this: 

```js
useTextBuffer({ 
  url: "api/chat",
  type: "chat" // defaults to "text" or "plaintext"
});
```
 
 Let me know what you think.

### PR DESCRIPTION
**Pull request by biowaffeln**
fixes #8

### CODE REVIEW

**Comment by ctjlewis** [comment]
Oh shit, this is good work. I'll review this in a few hours, sorry for the delay.

One thing more thing: Can you add a demo page? I can do it if you don't have time, but would prob be 3-5 days before I get to it. Either way we'll have to deploy it to Vercel production to test behavior. 

I kinda touched on this in the other issue but didn't explain fully: There's a divergence in how the chunks come back in Vercel production servers vs. on our dev machines. This is probably the biggest pain of the whole process, and you can only test actual prod behavior by deploying and verifying.

The short version is in dev our chunks always come back one at a time:

```
data: { a: 1, b: 2, c: 3 }
data: { a: 4, b: 5, c: 6 }
```

But in prod they come back as chunks of objects:

```
data: { a: 1, 
data: b: 2,
data: c: 3 }
...
```

**Comment by ctjlewis** [comment]
We will also apply this for the `chat` endpoint automatically rather than with a config option. `mode = "raw"` would get you today's output, just the stream of JSON-serialized objects as they come in (one object per chunk, this ^ must be done by us since production yields out partial objects).

It's needlessly complicated lol I appreciate your work and will add some stuff soon.

**Comment by daniel-black** [comment]
This was my approach to parsing the stream response from within a component: 

https://github.com/daniel-black/text-machina/blob/0aea18559dc7b131a8efd3a8dc11748a76c1fe64/src/components/chat/chat.tsx#L61

Occasionally in dev a chunk comes back with multiple chunks like: `'{"content":"I"}{"content":"see"}'` so my duct tape solution was to search the string for `"}{"` in the string and then split at the right index if found and parse both of the resulting chunks. Definitely not a bulletproof approach though, it errors pretty often when attempting to parse the chunks and I gotta dive into why when I have more time.

**Comment by biowaffeln** [comment]
@daniel-black Yeah I was pretty much doing the same thing with the regex and split. The code would've been a bit more elegant if I could use lookbehind regexes, but browser-support is currently too bad...

@ctjlewis just added a demo page under `/chat`, seems to work both in prod and dev here.

**Comment by joesaunderson** [comment]
Hey, do you know if this is close to being reviewed? Would love to use it!

**Comment by ctjlewis** [comment]
Super sorry for the delay on this guys, I've been stuck between a lot of stuff. **It should "just work" now** with (default) `tokens` mode, no need for custom flags in the buffer hooks as the ChatParser in `openai-streams` now unwraps the content.

@biowaffeln - I closed this as it was an upstream logic thing in streams, we were able to add this functionality with a lot fewer changes. I added your chat demo and it is the default demo now at `src/pages/index.tsx` (https://nextjs-openai.vercel.app). I made sure to give you co-author credit on changes + releases for both libs.

Thanks also @daniel-black for your notes.

Thanks everyone for being patient. If there are bugs, please report them.
